game/src/game.rs

```diff
@@ -77,83 +77,75 @@ impl Game {
         }
         Game { states, app }
     }
-}
-
-impl GUI for Game {
-    fn event(&mut self, ctx: &mut EventCtx) {
-        self.app.per_obj.reset();
 
-        let transition = self.states.last_mut().unwrap().event(ctx, &mut self.app);
-        // If we fall through, there's a new state that we need to wakeup.
+    // If true, then the top-most state on the stack needs to be "woken up" with a fake mouseover
+    // event.
```

Comment:
"fake mouseover" is this what the "no_op_event" is about?

Reply:
Yep. Probably could use a better name. I don't remember exactly the bug motivating this, but I think it was something like:
1) Click a button to enter some new state
2) Later, exit that state, but keep the mouse exactly in the same place
3) The button isn't in the hovering state until you generate a mouse event